### PR TITLE
@jonallured => Highlight matching parts of query

### PR DESCRIPTION
**Pull request by mzikherman**
screen shot 2019-02-12 at 1 58 09 pm

Uses a tiny lib that does a good of matching!

I think from a functional standpoint this is pretty much...complete(?!). Obviously there's more styling and UI work to do, but I/we can circle back to that.

Up next I think I'll try to bring this component in to Force (replacing the admin guarded placeholder added in https://github.com/artsy/force/pull/3428), and see how it looks 🤞

### CODE REVIEW

**Comment by artsyit** [comment]
Deploy preview for *artsy-reaction* ready!

Built with commit 673877e9045dd34f098085c64efda17b27fd91c0

https://deploy-preview-2012--artsy-reaction.netlify.com

File: src/Components/Search/__tests__/SearchBar.test.tsx [hunk 1]
```diff
@@ -93,4 +93,13 @@ describe("SearchBar", () => {
...
     expect(window.location.assign).toHaveBeenCalledWith("/cat/percy-z")
   })
+
+  it("highlights matching parts of suggestions", async () => {
+    const component = await getWrapper(searchResults)
+
+    simulateTyping(component, "perc") // Matching text w/ suggestion.
+    await flushPromiseQueue()
+
+    expect(component.html()).toContain("<strong>Perc</strong>y Z")
```
Comment (jonallured) [inline src/Components/Search/__tests__/SearchBar.test.tsx]: Very cool!! 😎

File: src/Components/Search/Suggestions/SuggestionItem.tsx [hunk 2]
```diff
@@ -1,18 +1,27 @@
...
   return (
-    <Box p={1}>
-      <Serif size={"2"}>{display}</Serif>
-      <Sans color={color("black30")} size={"2"}>
+    <Flex pl={3} pb={3} justifyContent="center" flexDirection="column">
+      <Serif size="2">
+        {parts.map(({ highlight, text }, index) => {
+          return highlight ? <strong key={index}>{text}</strong> : text
+        })}
+      </Serif>
```
Comment (jonallured) [inline src/Components/Search/Suggestions/SuggestionItem.tsx]: This is cool, but I think we'll need something like this for other parts of the Onsite Search 2.0 project, so it's possible we'll want to generalize this somehow. We shouldn't do anything about this now, but something to keep in mind!!

File: src/Components/Search/SearchBar.tsx [hunk 3]
```diff
@@ -78,7 +82,7 @@ export class SearchBar extends Component<Props, State> {
       },
       null,
       error => {
-        if (error) console.error(error)
+        if (error) logger.error(error)
```
Comment (jonallured) [inline src/Components/Search/SearchBar.tsx]: This seems out of scope for this PR - can you talk about what's going on here?

Comment (mzikherman) [reply src/Components/Search/SearchBar.tsx]: Just addressing some feedback from a prior PR- https://github.com/artsy/reaction/pull/2007#discussion_r255221517

This is the better way to do logging- it includes the file location and will also send to Sentry.

**Comment by artsyit** [comment]
:rocket: PR was released in v13.0.1 :rocket:
